### PR TITLE
Add .NoSharedFrameworkSchemes error related to #570

### PR DESCRIPTION
**Pull request by ikesyo**
Addresses https://github.com/Carthage/Carthage/issues/532#issuecomment-112984683, https://github.com/Carthage/Carthage/issues/532#issuecomment-121513872 and other comments.

If `.NoSharedSchemes` error happens, failover to other projects would not work well before the fix.

### CODE REVIEW

File: Source/CarthageKit/Xcode.swift [hunk 1]
```diff
@@ -992,6 +995,15 @@ public func buildInDirectory(directoryURL: NSURL, withConfiguration configuratio
 							|> map { _ in scheme }
 					}
 					|> collect
+					|> catch { error in
+						switch error {
+						case .NoSharedSchemes:
```
Comment (ikesyo) [inline Source/CarthageKit/Xcode.swift]: Should the `.XcodebuildListTimeout` error be handled also?

File: Source/CarthageKit/Xcode.swift [hunk 2]
```diff
@@ -992,6 +995,15 @@ public func buildInDirectory(directoryURL: NSURL, withConfiguration configuratio
...
+							return .empty
+
+						default:
+							return SignalProducer(error: error)
+						}
+					}
```
Comment (mdiep) [inline Source/CarthageKit/Xcode.swift]: What's the purpose of this `catch`?

Comment (ikesyo) [inline Source/CarthageKit/Xcode.swift]: > If `.NoSharedSchemes` error happens, failover to other projects would not work well before the fix.

Comment (mdiep) [inline Source/CarthageKit/Xcode.swift]: Do you mean that this lets Carthage continue to run after finding a project with no shared dependencies?

Comment (ikesyo) [inline Source/CarthageKit/Xcode.swift]: In other words, #570 works well if some projects does not have a shared framework scheme, but have some other shared schemes. If those projects have no any shared schemes, `buildInDirectory()` errors out at the first project locator of them.

File: Source/CarthageKit/Xcode.swift (hunk 1 above)
Comment (mdiep) [inline Source/CarthageKit/Xcode.swift]: No, I think it's best not to catch those.

**Comment by mdiep** [comment]
Looks good. :+1: Thanks!

**Comment by ikesyo** [comment]
:smile:

**Comment by ikesyo** [comment]
I'm sorry, this pull request breaks the compatibility with some repositories which have no projects/workspaces (e.g. https://github.com/jspahrsummers/xcconfigs). We might consider not to send `.NoSharedFrameworkSchemes` error if a dependency have no projects/workspaces.

**Comment by ikesyo** [comment]
A follow-up: #637. :bow:

**Pull request merged**